subsys/bluetooth/controller/ll_sw/ll_addr.c

```diff
@@ -81,6 +60,31 @@ uint8_t ll_addr_set(uint8_t addr_type, uint8_t const *const bdaddr)
 	return 0;
 }
 
+uint8_t *ll_addr_get(uint8_t addr_type)
+{
+	if (addr_type > BT_ADDR_LE_RANDOM) {
+		return NULL;
+	}
+
+	if (addr_type) {
+		return rnd_addr;
+	}
+
+	return pub_addr;
+}
+
+uint8_t *ll_addr_read(uint8_t addr_type, uint8_t *const bdaddr)
+{
+	uint8_t *addr;
+
+	addr = ll_addr_get(addr_type);
+	if (addr) {
+		memcpy(bdaddr, addr, BDADDR_SIZE);
+	}
+
+	return addr;
+}
```

Comment:
Correct me if I am wrong, but this function seems to return the address, as well as store it in the `bdaddr` argument, effectively "returning" it twice. 

Is this by design? It's similar to the memcpy/memset functions, so it's not necessarily wrong, just making sure it is intentional.

Reply:
yes, it is by design, was simpler to return a valid pointer or NULL, compared to returning a errno.